core/extension/gdextension.cpp

```diff
@@ -929,9 +929,13 @@ Error GDExtensionResourceLoader::load_gdextension_resource(const String &p_path,
 			compatible = VERSION_MAJOR < compatibility_maximum[0];
 		} else if (VERSION_MINOR != compatibility_maximum[1]) {
 			compatible = VERSION_MINOR < compatibility_maximum[1];
-		} else {
+		}
+#if VERSION_PATCH
+		// #if check to avoid -Wtype-limits warning when 0.
+		else {
 			compatible = VERSION_PATCH <= compatibility_maximum[2];
 		}
+#endif
```

Comment:
I don't think this is quite right. Even if `VERSION_PATCH` is zero, we still need to check that it matches `compatibility_maximum[2]`.

I think this should be something like:

```c++
#if VERSION_PATCH > 0
			compatible = VERSION_PATCH <= compatibility_maximum[2];
#else
			compatible = VERSION_PATCH == compatibility_maximum[2];
#endif
```

Reply:
Or, actually, maybe it is right! If `VERSION_PATCH` is zero, then it must be less than or equal to all values of `compatibility_maximium[2]`.

Reply:
Yes basically what the warning says is that this line is tautological when `VERSION_PATCH` is zero, since the other side is unsigned so will always be `>= 0`.

Reply:
Yeah, sorry, I was too tired to be reviewing PRs last night :-)